libsql-server/src/http/user/trace.rs

```diff
@@ -0,0 +1,103 @@
+use std::time::Duration;
+
+use hyper::{http, HeaderMap, Request, Response};
+use tonic::Status;
+use tower_http::{
+    classify::{
+        ClassifiedResponse, ClassifyResponse, GrpcCode, GrpcErrorsAsFailures, GrpcFailureClass,
+    },
+    trace::{DefaultOnResponse, OnResponse},
+};
+use tracing::{Level, Span};
+
+use crate::metrics::CLIENT_VERSION;
+
+pub(crate) fn request<B>(req: &Request<B>, span: &Span) {
+    let _s = span.enter();
+
+    tracing::debug!(
+        "got request: {} {} {:?}",
+        req.method(),
+        req.uri(),
+        req.headers()
+    );
+    if let Some(v) = req.headers().get("x-libsql-client-version") {
+        if let Ok(s) = v.to_str() {
+            metrics::increment_counter!(CLIENT_VERSION, "version" => s.to_string());
```

Comment:
that allocation bothers me still 😄

Reply:
Yeah, another option is to use some sort of static lookup table or update metrics to accept integers.